### PR TITLE
session: upgrade from old cluster infoschema v2 should not be used

### LINKED ISSUE #56544: upgrade from v7.5 to master, @@tidb_schema_cache_size incorrect
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Start a old version tidb cluster:

```
tiup playground v7.6.0 --tiflash 0
```

kill the tidb-server and restart a master tidb-server

### 2. What did you expect to see? (Required)

```
mysql> select @@global.tidb_schema_cache_size;
+---------------------------------+
| @@global.tidb_schema_cache_size |
+---------------------------------+
| 536870912                      |
+---------------------------------+
1 row in set (0.00 sec)

mysql> select * from GLOBAL_VARIABLES where variable_name = 'tidb_schema_cache_size';
Empty set (0.01 sec)
```

This means the global variable @@tidb_schema_cache_size does not exist in tikv.
So the default value 536870912 is used.

Then upgrade an old tidb cluster from v7.5 to master, it will use infoschema v2.

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: close #56544

Problem Summary:

### What changed and how does it work?

Upgrade from old version tidb, infoschema v2 should not be used.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [X] Manual test (add detailed scripts or steps below)

See the steps described in the issue.

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/session/bootstrap.go [hunk 1]
```diff
@@ -3267,6 +3272,15 @@ func upgradeToVer216(s sessiontypes.Session, ver int64) {
 	mustExecute(s, "UPDATE mysql.global_variables SET VARIABLE_VALUE='table' WHERE VARIABLE_NAME = 'tidb_scatter_region' AND VARIABLE_VALUE = 'ON'")
 }
 
+func upgradeToVer217(s sessiontypes.Session, ver int64) {
+	if ver >= version217 {
+		return
+	}
+	// If tidb_schema_cache_size does not exist, insert a record and set the value to 0
+	// Otherwise do nothing.
+	mustExecute(s, "INSERT IGNORE INTO mysql.global_variables VALUES ('tidb_schema_cache_size', 0)")
```
Comment (D3Hunter) [inline pkg/session/bootstrap.go:3281]: if there's no such row in `global_variables`

when we execute this statement, the first domain and infocache has already started, the infocache is still v2 ?

Comment (tiancaiamao) [reply pkg/session/bootstrap.go:3281]: Later when infoschema reload it would detect the new infoschema value and change to v1  @D3Hunter

**Review by D3Hunter — APPROVED** [review]

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Review by 3pointer — APPROVED** [review]

**Review by yudongusa — APPROVED** [review]
LGTM

**Pull request merged**